common/daemon_conn.c

```diff
@@ -73,23 +101,30 @@ static struct io_plan *daemon_conn_start(struct io_conn *conn,
 			 daemon_conn_write_next(conn, dc));
 }
 
-void daemon_conn_init(tal_t *ctx, struct daemon_conn *dc, int fd,
-		      struct io_plan *(*daemon_conn_recv)(struct io_conn *,
-							  struct daemon_conn *))
+static void destroy_dc_from_conn(struct io_conn *conn, struct daemon_conn *dc)
 {
-	dc->daemon_conn_recv = daemon_conn_recv;
-
-	dc->ctx = ctx;
-	dc->msg_in = NULL;
-	msg_queue_init(&dc->out, dc->ctx);
-	dc->msg_queue_cleared_cb = NULL;
-	dc->conn = io_new_conn(ctx, fd, daemon_conn_start, dc);
+	/* Harmless free loop if conn is being destroyed because dc freed */
+	tal_free(dc);
 }
 
-void daemon_conn_clear(struct daemon_conn *dc)
+struct daemon_conn *new_daemon_conn_(const tal_t *ctx, int fd,
```

Comment:
Keeping with the `ccan` naming convention this should be `daemon_conn_new_`.

Reply:
Yes.  Good point!